sdk/auth/managed_identity_authorizer.go

```diff
@@ -157,7 +157,11 @@ func (c *managedIdentityConfig) TokenSource(_ context.Context) (Authorizer, erro
 }
 
 func azureMetadata(ctx context.Context, url string) (body []byte, err error) {
-	ctx2, cancel := context.WithDeadline(ctx, time.Now().Add(time.Second*30))
+	const retryMaxCount = 5
+	const retryWaitMax = 60
+	const singleTimeout = 10
+	const timeout = retryMaxCount * (retryWaitMax + singleTimeout)
+	ctx2, cancel := context.WithDeadline(ctx, time.Now().Add(time.Second*(30+timeout)))
 	defer cancel()
```

Comment:
as mentioned in the last review, we shouldn't be using a timed context here (I appreciate this was already present, but should have been removed), since it hides the reliance on a timed context - we should instead be requiring that the `ctx` passed has a deadline attached:

```suggestion
	deadline, ok := ctx.Deadline()
	if !ok {
		return fmt.Errorf("internal-error: context had no deadline")
	}
```

Reply:
I got the following error with this code. Seems the deadline does not exist at this time. I think we may still need to set a deadline here. 
![image](https://user-images.githubusercontent.com/76987228/225289338-9ef306a4-f2ee-4b34-bac4-947a9f4ab028.png)

Reply:
Yep, we'll need to update the calling code too, that's expected given the new check